addons/google_drive/models/google_drive.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import re
+import urllib2
+import werkzeug
+
+from odoo import api, fields, models, _
+from odoo.exceptions import RedirectWarning, UserError
+from odoo.tools.safe_eval import safe_eval as eval
+from odoo.addons.google_account import TIMEOUT
+
+
+class GoogleDriveConfig(models.Model):
+    _name = 'google.drive.config'
+    _description = "Google Drive templates config"
+
+    name = fields.Char(string='Template Name', required=True)
+    model_id = fields.Many2one('ir.model', string='Model', ondelete='set null', required=True)
+    model = fields.Char(related='model_id.model', readonly=True)
+    filter_id = fields.Many2one('ir.filters', string='Filter', domain="[('model_id', '=', model)]")
+    google_drive_template_url = fields.Char(string='Template URL', required=True)
+    google_drive_resource_id = fields.Char(compute='_compute_resource_id', string='Resource ID')
+    google_drive_client_id = fields.Char(compute='_compute_client_id', string='Google Client')
+    name_template = fields.Char(string='Google Drive Name Pattern', required=True, default="Document %(name)s",
+                                help='Choose how the new google drive will be named, on google side. Eg. gdoc_%(field_name)s')
+    active = fields.Boolean(default=True)
+
+    @api.multi
+    def _compute_resource_id(self):
+        for config in self:
+            key = config._get_key_from_url(config.google_drive_template_url)
+            if key:
+                config.google_drive_resource_id = key
+            else:
+                raise UserError(_("Please enter a valid Google Document URL."))
+
+    @api.multi
+    def _compute_client_id(self):
```

Comment:
Is decoration of this method required(same thing applied to _compute_resource_id) ?